pkg/driver/screen/x11_linux.go

```diff
@@ -61,40 +60,16 @@ func (s *screen) Close() error {
 }
 
 func (s *screen) VideoRecord(p prop.Media) (video.Reader, error) {
-	rect := s.reader.img.Bounds()
-	w := rect.Dx()
-	h := rect.Dy()
-	imgI444 := image.NewYCbCr(rect, image.YCbCrSubsampleRatio444)
-
 	if p.FrameRate == 0 {
 		p.FrameRate = 10
 	}
 	s.tick = time.NewTicker(time.Duration(float32(time.Second) / p.FrameRate))
 
 	r := video.ReaderFunc(func() (image.Image, error) {
 		<-s.tick.C
-		img := s.reader.Read()
-		// Convert it to I444
-		imgI444.SubsampleRatio = image.YCbCrSubsampleRatio444
-		imgI444.Rect = rect
-		imgI444.YStride = w
-		imgI444.CStride = w
-		for y := 0; y < h; y++ {
-			iyBase := y * imgI444.YStride
-			icBase := y * imgI444.CStride
-			for x := 0; x < w; x++ {
-				iy := iyBase + x
-				ic := icBase + x
-				r, g, b, _ := img.At(x, y).RGBA()
-				yy, cb, cr := color.RGBToYCbCr(uint8(r/0x100), uint8(g/0x100), uint8(b/0x100))
-				imgI444.Y[iy] = yy
-				imgI444.Cb[ic] = cb
-				imgI444.Cr[ic] = cr
-			}
-		}
-		return imgI444, nil
+		return s.reader.Read(), nil
 	})
-	return r, nil
+	return video.ToI420(r), nil
```

Comment:
I think it's better to insert ToI420 just before the encoder. RGBA image is good at manipulation as the image processing libraries have optimized implementation for RGBA image.

Reply:
Ah right, that makes sense.

Reply:
So, I've updated it, but I got a panic, https://gist.githubusercontent.com/lherman-cs/02234dd460897baa520cffc111460fe2/raw/ccc32139b4a84b8dc4c80b2bc2cc8354d53051ac/gistfile1.txt.

Reply:
It looks like we have a bug in `vpx`. When I ran the example with `GODEBUG=cgocheck=2`, it crashed immediately and showed:

```
write of Go pointer 0xc000562000 to non-Go memory 0x12229e0

...
goroutine 27 [running]:
runtime.systemstack_switch()
        /usr/lib/go-1.13/src/runtime/asm_amd64.s:330 fp=0xc000073dc0 sp=0xc000073db8 pc=0x47b3d0
runtime.cgoCheckWriteBarrier(0x12229e0, 0xc000562000)
        /usr/lib/go-1.13/src/runtime/cgocheck.go:53 +0xc9 fp=0xc000073df8 sp=0xc000073dc0 pc=0x427ec9
runtime.wbBufFlush(0x12229e0, 0xc000562000)
        /usr/lib/go-1.13/src/runtime/mwbbuf.go:196 +0x83 fp=0xc000073e28 sp=0xc000073df8 pc=0x44caa3
runtime.gcWriteBarrier(0xc0002e97a0, 0xa11f60, 0xc00013d300, 0x0, 0x0, 0x0, 0x400, 0xc000346000, 0x99c301, 0x56293c, ...)
        /usr/lib/go-1.13/src/runtime/asm_amd64.s:1444 +0xae fp=0xc000073eb0 sp=0xc000073e28 pc=0x47d55e
github.com/pion/mediadevices/pkg/codec/vpx.(*encoder).Read(0xc000139720, 0xc000346000, 0x400, 0x400, 0x0, 0x0, 0x0)
        /home/lukas/Documents/pion/mediadevices/pkg/codec/vpx/vpx.go:135 +0x59a fp=0xc000073f78 sp=0xc000073eb0 pc=0x7c61ca
github.com/pion/mediadevices.(*videoTrack).start(0xc00033a180)
        /home/lukas/Documents/pion/mediadevices/track.go:139 +0x83 fp=0xc000073fd8 sp=0xc000073f78 pc=0x75cca3
runtime.goexit()
        /usr/lib/go-1.13/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc000073fe0 sp=0xc000073fd8 pc=0x47d4a1
created by github.com/pion/mediadevices.newVideoTrack
        /home/lukas/Documents/pion/mediadevices/track.go:130 +0x331
...
```

Reply:
It looks a pointer handling problem. https://github.com/pion/mediadevices/pull/60 should fix it.